app/src/main/java/com/humberto/tasky/agenda/presentation/agenda_details/AgendaDetailsViewModel.kt

```diff
@@ -283,8 +289,32 @@ class AgendaDetailsViewModel @Inject constructor(
         }
     }
 
+    private fun updateItem() {
+        applicationScope.launch {
+            _state.update {
+                it.copy(
+                    isSaving = true,
+                    isEditing = false
+                )
+            }
+            val agendaItem = _state.value.toAgendaItem()
+            when(agendaItem) {
+                is AgendaItem.Task -> taskRepository.updateTask(agendaItem)
+                is AgendaItem.Event -> eventRepository.updateEvent(agendaItem)
+                is AgendaItem.Reminder -> reminderRepository.updateReminder(agendaItem)
+            }.onSuccess {
+                alarmScheduler.cancelAlarm(agendaItem.id)
+                alarmScheduler.scheduleAlarm(agendaItem.toAlarmItem())
```

Comment:
Here, the developer needs to make sure to cancel the alarm before rescheduling it. Would it make more sense to instead check for an existing alarm and cancel it when scheduling a new alarm?